Add support for ed25519 keys in AWS

aclevername opened the pull request:
AWS EC2 supports `ed25519` keys from the start of this year [(blog)](https://aws.amazon.com/about-aws/whats-new/2022/01/ed25519-keys-authentication-ec2-instance-connect/), updating the `ComputeAWSKeyFingerprint` func to match this.

## Implementation notes
I trimmed the "SHA256" prefix that the `ssh.FingerprintSHA256` call adds to keep it consistent with the fingerprint format presented both in the console and the API. Happy to undo this if folks think this is incorrect :smile: 
![image](https://user-images.githubusercontent.com/19669095/155541904-2895ba51-4e8b-4739-ac4d-ae2b53db9bc3.png)

```
$ aws ec2 describe-key-pairs --key-names jake-test-ed25519
{
    "KeyPairs": [
        {
            ...
            "KeyFingerprint": "HvE7+gmH78VS53+iPuRDh/gKjVo26OzYU/qOnJWAgyk=",
            "KeyName": "jake-test-ed25519",
            "KeyType": "ed25519",
        }
    ]
}
```

This issue was originally discovered here https://github.com/weaveworks/eksctl/issues/4779

hakman commented:
/ok-to-test

olemarkus commented on pkg/pki/sshkey.go:
```diff
@@ -73,13 +73,19 @@ func ComputeAWSKeyFingerprint(publicKey string) (string, error) {
+		h := md5.Sum(der)
+		return colonSeparatedHex(h[:]), nil
+	case "ssh.ed25519PublicKey":
+		return strings.TrimPrefix(ssh.FingerprintSHA256(sshPublicKey), "SHA256:"), nil
```
Is there a reason for using SHA256 for ed keys, but MD5 for RSA?
Perhaps use MD5 for both, optionally add a flag for hash type (like `ssh-add -l` does).

aclevername replied:
Just following the same method the AWS API use to calculate the fingerprint. I added an example in the PR description 😃

olemarkus replied:
Fair enough. I think we should keep the prefix so we know what hash algo is actually used.

olemarkus reviewed (approved):
Awesome. Thanks for this. Feel free to cherry-pick this to the 1.23 branch.

olemarkus commented:
/hold
Looks like you may need a rebase or something.

aclevername commented:
> /hold Looks like you may need a rebase or something.

already have @olemarkus  :thinking: 
```
-> % git pull origin master
From github.com:kubernetes/kops
 * branch                  master     -> FETCH_HEAD
Already up to date.

-> % git co ed25519
Switched to branch 'ed25519'
Your branch is up to date with 'aclevername/ed25519'.

-> % git rebase origin/master
Successfully rebased and updated refs/heads/ed25519.

-> % git push --force
Everything up-to-date

-> % git remote -v
aclevername     git@github.com:aclevername/kops.git (fetch)
aclevername     git@github.com:aclevername/kops.git (push)
origin  git@github.com:kubernetes/kops.git (fetch)
origin  git@github.com:kubernetes/kops.git (push)
```

olemarkus commented:
/retest

aclevername commented:
> Awesome. Thanks for this. Feel free to cherry-pick this to the 1.23 branch.

Thanks, just to clarify you mean open an identical PR to merge into the `release-1.23` branch? Are older versions also open for back patching? We currently use 1.21 which would be great to have it in

olemarkus commented:
/hold cancel

flake it seems
